addons/web/static/tests/boot_tests.js

```diff
@@ -1,18 +1,66 @@
-(function () {
+(async function () {
 "use strict";
 
-odoo.__DEBUG__.didLogInfo.then(function () {
-    var modulesInfo = odoo.__DEBUG__.jsModules;
+    // We define here a test asserting that no module failed to be loaded. The
+    // module system (see boot.js) waits 10 seconds before logging that some
+    // modules couldn't be loaded, when it happens. The test below is really
+    // important as it will detect testing modules that can't be loaded, for
+    // instance because one of their dependencies is missing. Without it, we
+    // could have tests that aren't executed, without any notice.
+    //
+    // We have the following constraints around this test:
+    //  - we want it to be always executed (obviously)
+    //  - we need it to wait for the module system to complete the loading of
+    //    the modules before being executed (wait for the didLogInfo promise)
+    //  - we don't want it to slow down the whole suite
+    //  - we can't rely on the module system to define this test, as it is
+    //    partially testing it as well
+    //
+    // To achieve this, we ensure that this test is defined after (some of) the
+    // other tests. Almost all tests need module 'web.test_utils', which is
+    // asynchronous (it loads resources that are lazy-loaded outside the test
+    // environment). So here, we also wait for 'web.test_utils' (without using
+    // the module system) before defining our test, so that our test is defined
+    // after all those requiring 'web.test_utils'. That way:
+    //  - either the suite takes more than 10 seconds to execute, and once we
+    //    reach our test, the promise is already resolved and it can directly
+    //    check if some modules failed
+    //  - or the suite takes less than 10 seconds, and our test will wait for
+    //    the promise to be resolved.
+    // However, in any cases, it won't slow down the test suite longer than
+    // necessary.
+    let resolve;
+    const prom = new Promise(r => {
+        resolve = r;
+    });
```

Comment:
Even if other suggestions are not done, the `setInterval` part should still be wrapped inside this promise constructor to avoid having to extract the `resolve`. This will also allow the promise to properly reject if there is an exception inside that part of the code.

Reply:
totally agree. I spent more time on the explanation than on the coding style :D